torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -466,63 +522,83 @@ def _export_fx_node_to_onnxscript(
                 )
             current_attr = getattr(current_attr, sub_attr_name)
 
-        input_ = onnxscript_graph.add_initializer(node.name, current_attr)
+        input_ = self.onnxscript_graph.add_initializer(node.name, current_attr)
 
-        assert isinstance(input_, graph_building.TorchScriptTensor)
+        assert isinstance(input_, onnxscript_graph_builder.TorchScriptTensor)
         assert isinstance(input_, onnxscript.tensor.Tensor)
-        fx_name_to_onnxscript_value[node.name] = input_
+        self.fx_name_to_onnxscript_value[node.name] = input_
         # FIXME: Refactor logic getting 'current_attr'.
         assert isinstance(current_attr, torch.Tensor)
-    else:
-        # TODO(wechi): Support get_attr, call_module, call_method.
-        raise RuntimeError(f"Found node type not defined in torch.fx: {node.op}")
 
 
-@_beartype.beartype
-@diagnostics.diagnose_call(diagnostics.rules.atenlib_fx_to_onnx)
-def export_fx_to_onnxscript(
```

Comment:
I think in this PR, we just need
```python
class Interpreter:
  @staticmethod
  def export_fx_to_onnxscript(...)
  @staticmethod
  def _export_fx_node_to_onnxscript(...)
```
Then you don't need those changes for making these functions a stateful class. It will be much cleaner --- interpreter-related logics are in this class but in a stateful way as it was.

Reply:
Extending this a little, after they become static methods, them being in the class is purely for grouping/namespacing. 

Now that we have a nice module to host this logic, putting the functions in the module global namespace is a preferred way to organize them:

> Never use staticmethod unless forced to in order to integrate with an API defined in an existing library. Write a module-level function instead.

https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators

Reply:
Yep, if these functions can be static, they should just live in the module. What I care about is the locality of these things so we don't have spaghetti in a generic until module. Pure functions in sensibly organized modules is great.

Reply:
> I think in this PR, we just need
> 
> ```python
> class Interpreter:
>   @staticmethod
>   # Rename it to export_graph maybe
>   def export_fx_to_onnxscript(...)
>   @staticmethod
>   # Rename it to export_node maybe
>   def _export_fx_node_to_onnxscript(...)
>   # Other interpreter-specific utils.
>   ...
> ```
> 
> Then you don't need those changes for making these functions a stateful class. It will be much cleaner --- interpreter-related logics are in this class but in a stateful way as it was.

then you would have a stateful function. that is not any better than stateful method, right?

Reply:
A function is pure (stateless) if the outputs are solely determined by its inputs. I don't think taking contexts makes it stateful?

Reply:
> Yep, if these functions can be static, they should just live in the module. What I care about is the locality of these things so we don't have spaghetti in a generic until module. Pure functions in sensibly organized modules is great.

I will double check on that. But Wei-Shen idea implies all FX Nodes will be implemented within a huge function with if-else. The current approach segregates each FX node into its own method

Reply:
> 

`Shen idea implies all FX Nodes will be implemented within a huge function with if-else` --- It's ok. It explicitly describes how a `fx.Node` should be re-routed. If the routing logics are hidden by a class and `self` attribute, it will be harder to understand the already-complicated logics.

Reply:
> A function is pure (stateless) if the outputs are solely determined by its inputs. I don't think taking contexts makes it stateful?

Totally agree. Passing context means context is swappable and allows easier hacking.

Reply:
> all FX Nodes will be implemented within a huge function with if-else. The current approach segregates each FX node into its own method

This can be achieved by creating pure function for each optype without involving class refactoring.

The design for `torch.fx.Interpreter` trades statelessness with simpler user interface for subclassing UX. They intentionally hide away common state maintenance details, such that it is simple to use for light-weight tasks, like replacing all dropout op with noop. But the complexity immediately crawls back and in-depths knowledge required once you want to do something more advanced.

The target use case does not fit here because `fx_to_onnxscript` is a one-off "pass". We still need to create our own state management logic, and that logic is not re-used anywhere. We only do fx to onnx once.

Reply:
@BowenBao, let's keep this PR simple and not to refactorize the code. I know you meant the if-elif-else for different `fx.Node..op_type`. But I feel it clearer to list all cases inside a single place. Since purely moving and renaming will fix enough problems, I prefer to defer refactorizations to other PRs.

Reply:
@wschin Is there a settlement on whether to make the optype methods stateless or revert them back to single method? I might be missing some discussion, but I'm fine with either decision.

Reply:
Will create a stateless version on a per-optype basis soon